cookbook/case_studies/ml_training/house_price_prediction/house_price_predictor.py

```diff
@@ -96,38 +120,43 @@ def gen_houses(num_houses) -> pd.DataFrame:
 
 
 # %%
-# Split the data into train, val, and test datasets.
+# Data Preprocessing
+# =====================
+#
+# We will create `feature` and `target` columns and split them into train and test subsets.
+# We will also split the input DataFrame into train, validation, and test datasets.
 def split_data(
     df: pd.DataFrame, seed: int, split: typing.List[float]
 ) -> Tuple[pd.DataFrame, pd.DataFrame, pd.DataFrame]:
 
     seed = seed
-    val_size = split[1]
-    test_size = split[2]
+    val_size = split[1]  # 0.3 
+    test_size = split[2] # 0.1
 
     num_samples = df.shape[0]
-    x1 = df.values[
-        :num_samples, 1:
-    ]  # keep only the features, skip the target, all rows
-    y1 = df.values[:num_samples, :1]  # keep only the target, all rows
+    # retain the features, skip the target column
+    x1 = df.values[:num_samples, 1:]  
+    # retain the target column
+    y1 = df.values[:num_samples, :1]  
 
-    # Use split ratios to divide up into train & test
+    # divide the features and target columns into random train and test subsets, based on `test_size` 
     x_train, x_test, y_train, y_test = train_test_split(
         x1, y1, test_size=test_size, random_state=seed
     )
-    # Of the remaining training samples, give proper ratio to train & validation
+    # divide the input data into train and validation subsets, based on `test_size`
```

Comment:
```suggestion
    # divide the train data into train and validation subsets, based on `test_size`
```

Reply:
shouldn't it be 'input' data? we divide this input data into train, test and validation data  right?

Reply:
Input basically is train data in our case. Correct?

Reply:
doesn't it sound different to say we divide train data into train test and validation data?

Reply:
You are right, it has to be train data.